dash/include/dash/matrix/internal/LocalMatrixRef-inl.h

```diff
@@ -321,14 +323,21 @@ ::at(
       "expected " << (NumDim - _refview._dim) << " " <<
       "got " << sizeof...(Args));
   }
-  std::array<index_type, NumDim> coord = { static_cast<index_type>(args)... };
+  std::array<index_type, NumDim> coord = {
+    static_cast<index_type>(args)...
+  };
   for(auto i = _refview._dim; i < NumDim; ++i) {
     _refview._coord[i] = coord[i-_refview._dim];
   }
+  DASH_LOG_TRACE("LocalMatrixRef.at()",
+                 "ndim:",     NumDim,
+                 "curdim:",   CUR,
+                 "l_coords:",   _refview._coord,
+                 "l_viewspec:", _refview._l_viewspec);
   return local_at(
            _refview._mat->_pattern.local_at(
              _refview._coord,
-             _refview._viewspec));
+             _refview._l_viewspec));
```

Comment:
Supposedly this change is all that is required to fix the defect, yes.
`_coord` refers to local index range, `_viewspec` to global index range (global view to local index range, that is). Should be viewspec in local index range, of course. Fixed tests on my machines.

Also added unit test to cover this use case.